index.js

```diff
@@ -42,13 +47,32 @@ const makeEntityMapKeys = (sys) =>
  * @return {String}
  */
 const lookupInEntityMap = (entityMap, linkData) => {
-  const { entryId, linkType, spaceId } = linkData
-  if (spaceId) {
-    return entityMap.get(`${spaceId}!${linkType}!${entryId}`)
+  const { entryId, linkType, spaceId, environmentId } = linkData
+
+  if (spaceId && environmentId) {
+    return entityMap.get(`${spaceId}!${environmentId}!${linkType}!${entryId}`)
   }
+
   return entityMap.get(`${linkType}!${entryId}`)
 }
 
+const getIdsFromUrn = (urn) => {
+  const regExp = /.*:spaces\/([A-Za-z0-9]*)\/entries\/([A-Za-z0-9]*)/
```

Comment:
instead of using 2 regular expressions, use only one while making the `/environments/<environmentId>` optional, just like we did in the `resource-names` package [here](https://github.com/contentful/resource-names-stack/blob/3d7d819b8125ef999c659916f272ba12f3a0f94c/packages/resource-names/lib/constants.ts#L2C21-L2C21)